src/Control/Monad/Aff.js

```diff
@@ -96,6 +96,26 @@ exports._forkAff = function (nonCanceler, aff) {
   };
 }
 
+exports._forkAll = function (nonCanceler, foldl, affs) {
+  var voidF = function(){};
+
+  return function(success, error) {
+    foldl(function(_) {
+      return function(aff) {
+        aff(voidF, voidF);
+      };
+    })({})(affs);
+
+    try {
+      success({});
+    } catch(e) {
+      error(e);
+    }
+
+    return nonCanceler;
```

Comment:
A canceller is, I believe, an `Aff` which contains a function which takes an error and returns an `Aff` of a boolean. So yes, we'd collect all the cancellers in the fold, and then do some really messy logic (in JS, anyway) to propagate the cancel signal to the cancellers, and then back-propagate the booleans from the sub-cancellers to the parent canceller. I think you can find examples of this elsewhere in the JS code.